Nef_3/include/CGAL/Nef_3/SNC_simplify.h

```diff
@@ -544,7 +544,9 @@ class SNC_simplify_base : public SNC_decorator<SNC_structure> {
   void create_boundary_links_forall_sfaces(
       Unique_hash_map< SFace_handle, UFH_sface>& hash,
       Union_find< SFace_handle>& uf ) {
-    Unique_hash_map< SHalfedge_handle, bool> linked(false);
+    Unique_hash_map< SHalfedge_handle, bool> linked(false, this->sncp()->number_of_shalfedges());
+    this->sncp()->reserve_sm_boundary_items(this->sncp()->number_of_sfaces());
```

Comment:
This is the offending line. I should have picked this up earlier, but when performance testing I disable cgal assertions, and forgot to re-enable. This line should be removed as it cannot work currently.

Reply:
So unless you rewrite the rehash immediately, please remove the offending line.